shell/platform/android/io/flutter/embedding/android/FlutterViewDelegate.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.android;
+
+import android.app.Activity;
+import android.content.Context;
+import android.graphics.Rect;
+import android.view.Window;
+import android.view.WindowInsets;
+import androidx.annotation.RequiresApi;
+import androidx.annotation.VisibleForTesting;
+import io.flutter.Build;
+import io.flutter.util.ViewUtils;
+import java.util.Collections;
+import java.util.List;
+
+/** A delegate class that performs the task of retrieving the bounding rect values. */
+public class FlutterViewDelegate {
+  /**
+   * Return the WindowInsets object for the provided Context, or null if there is no associated
+   * activity.
```

Comment:
nit: Comment is out of date, returns null when any of
1. no associated activity
2. the window is null
3. `getRootWindowInsets()` returns null